mockrxandroidble/src/test/groovy/com/polidea/rxandroidble2/mockrxandroidble/RxBleClientMockTest.groovy

```diff
@@ -356,28 +389,221 @@ public class RxBleClientMockTest extends ElectricSpecification {
                 .take(1)
                 .map { scanResult -> scanResult.getBleDevice() }
                 .flatMap { rxBleDevice -> rxBleDevice.establishConnection(false) }
-                .flatMapSingle { rxBleConnection -> rxBleConnection.readCharacteristic(characteristicUUID) }
+                .flatMapSingle { rxBleConnection -> rxBleConnection.readCharacteristic(characteristicUUIDNoCallback) }
                 .map { data -> new String(data) }
                 .test()
 
         then:
         testSubscriber.assertValue("Polidea")
     }
 
+    def "should return characteristic data via callback"() {
+        when:
+        characteristicReadSubject.take(1).subscribe { result -> result.success("PolideaCB".getBytes()) }
+        def testSubscriber = rxBleClient.scanBleDevices(new ScanSettings.Builder().build())
+                .take(1)
+                .map { scanResult -> scanResult.getBleDevice() }
+                .flatMap { rxBleDevice -> rxBleDevice.establishConnection(false) }
+                .flatMapSingle { rxBleConnection -> rxBleConnection.readCharacteristic(characteristicUUID) }
+                .map { data -> new String(data) }
+                .test()
```

Comment:
This test seems to be testing a bit more than only if the callback is behaving correctly (scanning and connecting as well). We could probably cut the scanning part (as it is already tested in a different test) and start using `mockRxBleDevice` directly (as other test would prove that it is returned from scan). WDYT?